python/rapidsmpf/rapidsmpf/integrations/dask/shuffler.py

```diff
@@ -70,29 +68,32 @@ def get_shuffler(
     Notes
     -----
     Whenever a new :class:`Shuffler` object is created, it is
-    saved as ``dask_worker._rmp_shufflers[shuffle_id]``.
+    saved as ``DaskWorkerContext.shufflers[shuffle_id]``.
 
     This function is expected to run on a Dask worker.
     """
-    dask_worker = dask_worker or get_worker()
-    with get_worker_thread_lock():
-        if shuffle_id not in dask_worker._rmp_shufflers:
+    ctx = get_worker_context(dask_worker)
+    with ctx.lock:
+        if shuffle_id not in ctx.shufflers:
             if partition_count is None:
                 raise ValueError(
                     "Need partition_count to create new shuffler."
                     f" shuffle_id: {shuffle_id}\n"
-                    f" Shufflers: {dask_worker._rmp_shufflers}"
+                    f" Shufflers: {ctx.shufflers}"
                 )
-            dask_worker._rmp_shufflers[shuffle_id] = Shuffler(
-                get_comm(dask_worker),
-                get_progress_thread(dask_worker),
+            assert ctx.br is not None
+            assert ctx.comm is not None
+            assert ctx.progress_thread is not None
+            ctx.shufflers[shuffle_id] = Shuffler(
```

Comment:
*question*: I see this is not handled at the moment anyway, but whose job is it to clean up these created shufflers once we are done with them?

I note that shuffle ids are just a continuously incrementing integer, but `op_id` in the shuffler is an `unsigned char` so there can only be 256 concurrent shuffles (which is probably fine). But we _should_ release ids once we are done with a shuffle.

And ideally we would restrict this `ctx.shufflers` to only have 256 slots so we can detect and produce an error if we ever go over the 256 concurrent shuffle limit

Reply:
I don't think we even support 256 consecutive shuffles.
Let's fix this in a new PR: https://github.com/rapidsai/rapidsmpf/issues/239